add twentyThreeVideo template/slide

kasperbirch1 opened the pull request:
NYOSD-14

danquah commented on docker-compose.override.yml:
```diff
@@ -0,0 +1,8 @@
+  nginx:
+    labels:
+      # Enable https for the default route
+      - "traefik.http.routers.${COMPOSE_PROJECT_NAME}.tls=true"
```
Should be added to .gitignore

danquah commented on docker-compose.yml:
```diff
@@ -24,7 +24,9 @@ services:
       - "traefik.http.routers.${COMPOSE_PROJECT_NAME}.rule=Host(`${COMPOSE_DOMAIN}`)"
-
+      #      HTTPS config - uncomment to enable redirect from :80 to :443
+      - "traefik.http.routers.${COMPOSE_PROJECT_NAME}.middlewares=redirect-to-https"
```
Discard this chunck - and if we needed it it should go into the override-file

danquah commented on src/twenty-three-video/README.md:
```diff
@@ -0,0 +1,35 @@
+# TwentyThreeVideo Notes:
+
+## Url Prams
+https://www.twentythree.com/help/embedding-your-videos
```
Good idéa to add the link - but you should probably also add some text describing when the reader should consult the link.

danquah commented on src/twenty-three-video/README.md:
```diff
@@ -0,0 +1,35 @@
+https://www.twentythree.com/help/embedding-your-videos
+
+
+## player events
```
"Player Events"

Also, I would expect the section to then describe the various event we handle.

danquah commented on src/twenty-three-video/README.md:
```diff
@@ -0,0 +1,35 @@
+      function (e) {
+        console.log(e);
+        if (e.data.includes("player:video:ended")) {
+          if (lastVideoInPlaylist) return setVideoId(videoList[0]);
```
I would do something like this (pseudocode)
```javascript
let currentVideoIndex=0;
if (e.data.includes("player:video:ended")) {
  if (videoList.length <= currentVideoIndex+2) {
    // Go to the next slide
  } else {
    // Advance to the next video.
   currentVideoIndex = currentVideoIndex + 1;
  setVideoId(videoList[currentVideoIndex])
  }
 .. etc
}
```

danquah commented on src/twenty-three-video/twenty-three-video.js:
```diff
@@ -0,0 +1,113 @@
+//# sourceURL=twenty-three-video.js
+import React, { useEffect, useState } from "react";
+// import "./twenty-three-video.scss";
```
Remove commented lines

danquah commented on src/twenty-three-video/twenty-three-video.js:
```diff
@@ -0,0 +1,113 @@
+ */
+function TwentyThreeVideo({ slide, content = {}, run, slideDone }) {
+  // Content from content
+  const { videoList = "69389106, 74774561" } = content;
```
The default value should not contain any video ids

danquah commented on src/twenty-three-video/twenty-three-video.js:
```diff
@@ -0,0 +1,113 @@
+  // Content from content
+  const { videoList = "69389106, 74774561" } = content;
+
+  const formattedVideoList = videoList.replaceAll(".", ",").split(",").map((item) => item.trim()).filter(Boolean);
```
Would be nice with a comment so that I dont have to deconstruct the line.

danquah commented on src/twenty-three-video/twenty-three-video.js:
```diff
@@ -0,0 +1,113 @@
+    window.addEventListener(
+      "message",
+      function (e) {
+        if (e.data.includes("player:video:ended")) {
```
See previous comment

danquah commented on src/twenty-three-video/twenty-three-video.js:
```diff
@@ -0,0 +1,113 @@
+    return function cleanup() {
+      window.removeEventListener("message",
+        (e) => {
+          console.log("🚀 ~ cleanup", e)
```
Your removing the wrong event listener - you should use the const mentioned above

danquah commented on src/twenty-three-video/twenty-three-video.js:
```diff
@@ -0,0 +1,113 @@
+
+    window.addEventListener(
+      "message",
+      function (e) {
```
Add this function to a const first, then pass it to addEventListener so that you have a reference you can pass when you remove it below.

danquah commented on src/twenty-three-video/twenty-three-video.js:
```diff
@@ -0,0 +1,113 @@
+    if (!videoId) return <div className="">No ID</div>;
+    return (
+      <iframe
+        id="myAppIframe"
```
You need to generate a unique ID here - take a look at the slide your component is passed as a prop and see if you can determine a unique id from that - preferable something that includes the id of the region as you may have the same slide running in multiple regions.

kasperbirch1 committed:
add twentyThreeVideo template/slide

this slide can display 23 video from a comma separated list and will switch to the next slide after the last v ideo in the list has been played

NYOSD-14

kasperbirch1 committed:
Build slides

kasperbirch1 committed:
add docker-compose.override.yml to .gitignore

kasperbirch1 committed:
use id from slide to iframeID
